solutions/devsprint-riccardo-1/OnboardingChallenge/Screens/Settings/Views/SettingsViewController.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  SettingsViewController.swift
+//  OnboardingChallenge
+//
+//  Created by Rodrigo Borges on 03/11/21.
+//
+
+import UIKit
+
+let NUMBEROFROWSINSECTION = 1
+
+protocol SettingsViewDisplayLogic {
+    func display(viewModel: SettingsViewConfiguration.Get.ViewModel)
+}
+
+class SettingsViewController: UIViewController {
+    
+    private lazy var settingsView: SettingsView = {
+        return SettingsView()
+    }()
+    
+    var interactor: SettingsBusinessLogic?
+    var presenter: SettingsPresentationLogic?
+        
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        let request = SettingsViewConfiguration.Get.Request(key: "CFBundleVersion")
+        interactor?.getAppVersion(request: request)
+    }
+    
+    override func loadView() {
+        self.view = self.settingsView
+    }
+    
+    init() {
+        super.init(nibName: nil, bundle: nil)
+        setup()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func setup() {
+        let viewController = self
+        let interactor = SettingsInteractor(bundle: Bundle.main)
```

Comment:
Here we have the same issues we use to have in the other class. We are using the `Bundle.main` directly. How can we test the `setup` method if it's executed in a different target? 
We need to receive the bundle as a dependency from the outside world.

The best approach would be to request the same abstraction requested by the `SettingsInteractor` instead of a Bundle object.

Reply:
I initialize the interactor inside the ViewController however I am thinking if I could implement it so the Architecture is : 
ViewController -> ViewModel -> Interactor
Interactor -> ViewModel -> ViewController